File: WebContent/WEB-INF/jython/plugintopic_query.py
from cn.edustar.jitar.util import ParamUtil
from cn.edustar.jitar.pojos import User
from cn.edustar.jitar.model import ObjectType
from cn.edustar.jitar.service import PageKey
from cn.edustar.jitar.util import CommonUtil
from base_action import BaseAction
from cn.edustar.jitar.data import BaseQuery
from base_action import *

class PlugInTopicQuery(BaseQuery):
    def __init__(self, selectFields):
        BaseQuery.__init__(self, selectFields)
        self.params = ParamUtil(request)
        self.parentGuid = None
        self.parentObjectType = None       
        self.createUserId = None
        self.orderType = 0
        self.k = self.params.getStringParam("k")
        request.setAttribute("k", self.k)        
        
    def initFromEntities(self, qctx):
        qctx.addEntity("PlugInTopic", "pt", "")
             
    def resolveEntity(self, qctx, entity):
        BaseQuery.resolveEntity(self, qctx, entity)
                   
    def applyWhereCondition(self, qctx):
        if (self.parentGuid != None):
            qctx.addAndWhere("pt.parentGuid = :parentGuid")
            qctx.setString("parentGuid", self.parentGuid)
        if (self.parentObjectType != None):
            qctx.addAndWhere("pt.parentObjectType = :parentObjectType")
            qctx.setString("parentObjectType", self.parentObjectType)
        if (self.createUserId != None):
            qctx.addAndWhere("pt.createUserId = :createUserId")
            qctx.setInteger("createUserId", self.createUserId)
            
    def applyOrderCondition(self, qctx):
        if self.orderType == 0:
            qctx.addOrder("pt.plugInTopicId DESC")            

class PlugInTopicReplyQuery(BaseQuery):
    def __init__(self, selectFields):
        BaseQuery.__init__(self, selectFields)
        self.params = ParamUtil(request)
        self.plugInTopicId = None
        self.createUserId = None
        self.orderType = 1
        
    def initFromEntities(self, qctx):
        qctx.addEntity("PlugInTopicReply", "ptr", "")        
             
    def resolveEntity(self, qctx, entity):
        if "pt" == entity:
            qctx.addEntity("PlugInTopic", "pt", "pt.plugInTopicId = ptr.plugInTopicId")
        else:
            BaseQuery.resolveEntity(self, qctx, entity)
                   
    def applyWhereCondition(self, qctx):
        if (self.plugInTopicId != None):
            qctx.addAndWhere("ptr.plugInTopicId = :plugInTopicId")
            qctx.setInteger("plugInTopicId", self.plugInTopicId)
        if (self.createUserId != None):
            qctx.addAndWhere("ptr.createUserId = :createUserId")
            qctx.setInteger("createUserId", self.createUserId)        
            
    def applyOrderCondition(self, qctx):
        if self.orderType == 0:
            qctx.addOrder("ptr.plugInTopicReplyId DESC")
            
        if self.orderType == 1:
            qctx.addOrder("ptr.plugInTopicReplyId ASC")

File: src/cn/edustar/jitar/service/PlugInTopicQuery.java
package cn.edustar.jitar.service;

import cn.edustar.jitar.data.BaseQuery;
import cn.edustar.jitar.data.QueryContext;

public class PlugInTopicQuery extends BaseQuery {

	public String parentGuid = null;
	public String parentObjectType = null;
	public Integer createUserId = null;
	public Integer orderType = 0;

	public PlugInTopicQuery(String selectFields) {
		super(selectFields);
	}

	@Override
	public void initFromEntities(QueryContext qctx) {
		qctx.addEntity("PlugInTopic", "pt", "");
	}

	@Override
	public void applyWhereCondition(QueryContext qctx) {
		if (parentGuid != null) {
			qctx.addAndWhere("pt.parentGuid = :parentGuid");
			qctx.setString("parentGuid", parentGuid);
		}
		if (parentObjectType != null) {
			qctx.addAndWhere("pt.parentObjectType = :parentObjectType");
			qctx.setString("parentObjectType", parentObjectType);
		}
		if (createUserId != null) {
			qctx.addAndWhere("pt.createUserId = :createUserId");
			qctx.setInteger("createUserId", createUserId);
		}
	}

	@Override
	public void applyOrderCondition(QueryContext qctx) {
		if (orderType == 0) {
			qctx.addOrder("pt.plugInTopicId DESC");
		}
	}

	@Override
	public void resolveEntity(QueryContext qctx, String entity) {
		super.resolveEntity(qctx, entity);
	}

	public void setParentObjectType(String parentObjectType) {
		this.parentObjectType = parentObjectType;
	}

	public void setCreateUserId(Integer createUserId) {
		this.createUserId = createUserId;
	}

	public void setOrderType(Integer orderType) {
		this.orderType = orderType;
	}

	public void setParentGuid(String parentGuid) {
		this.parentGuid = parentGuid;
	}
}
